src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -330,6 +330,21 @@ public void timeEvent(final String eventName) {
     }
 
     /**
+     * Map version of the track API below.
+     *
+     * @param eventName The name of the event to send.
+     * @param properties A Map containing the key value pairs of the properties to include in this event.
+     *                   Pass null if no extra properties exist. Cannot be null.
+     */
+    public void trackMap(String eventName, Map<String, Object> properties) {
+        if (null == properties) {
+            Log.e(LOGTAG, "trackMap does not accept null properties");
+            return;
+        }
+        track(eventName, new JSONObject(properties));
```

Comment:
it might throw NullPointerException, but I guess we want to crash the app in that case?